lib/oli/delivery/attempts/core.ex

```diff
@@ -589,6 +589,26 @@ defmodule Oli.Delivery.Attempts.Core do
     |> Repo.preload(revision: [:activity_type])
   end
 
+  def get_latest_activity_attempt(section_id, user_id, resource_id) do
+    Repo.one(
```

Comment:
This does not take into account that a resource (page) can have multiple attempts.  This should be reworked to ensure that you are fetching the latest attempt for this activity WITHIN the latest attempt for the page.

Reply:
Not sure I understand this. From what I tell, the query fetches the last activity attempt based on the greater activity attempt id, which means that for a particular activity, user and section combo, it will get the greatest activity attempt regardless of what page attempt that belongs to.